nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -187,6 +187,19 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
             augmentor = None
 
         shuffle = config['shuffle']
+        device = 'gpu' if torch.cuda.is_available() else 'cpu'
```

Comment:
Is it the right assumption to always use gpu if available for dali?

Reply:
This is copied from the ctc_model.py. It's not exactly correct - you may have a model on cpu but GPU is available, but for training this is a good enough assumption.